### ISSUE TITLE
Calls REST API to provide max number of concurrent call per account/client per day per month

### ISSUE THREAD

**Issue by gvagenas** [RestComm/Restcomm-Connect#1590]

**Comment by forever3000** [comment]
Hi @gvagenas ,

I'm still not clear about max number of concurrent call. Like this example:
At 9:00AM, there are 2 concurrent call on Alice; At 10:AM there are 5 concurrent call; At 11:00AM there are 4 concurrent call so we should return 5 in this case ? Or concurrent call is a pre-definition. 
Same as this issue https://github.com/RestComm/Restcomm-Connect/issues/1590, what value should be returned in this case.
Was the number of concurrent call per account already implemented on sip servlet or we must do it manually based on the status of each call.

Thanks and Best Regards,
Vu Nguyen

**Comment by gvagenas** [comment]
Hi @forever3000 thanks for contributing in this issue.
You are right about the definition of concurrent calls. We should count maximum number of concurrent calls per account and/or per client. 
Also maybe you can have three kind of counters, one for the last hour, one for last day and one for specific day.

Check MonitoringService that already has counters for concurrent calls (global counter).

The response should be in line with the Call REST API response. Maybe the response should be part of the response when we GET on the CALLS API.

**Comment by forever3000** [comment]
Hi @gvagenas ,

Thank for your clear detail. I have just back after a business trip so sorry for no progress. However as I see, currently the max concurrent call (global counter) is not saved to database so it will be reset once server restart? 
And all call details have been stored in callDetailsMap hash map so may I want to count last hour, last day or specific day concurrent call per account or client by filtering in callDetailsMap hash map?

Thanks and Best Regards,
Vu Nguyen

**Comment by gvagenas** [comment]
@forever3000 you are right, the counters are cleared on instance restart and we don't store them in database. 
CDR records should be used to calculate the metrics. In a cluster environment where multiple Restcomm instances share the same DB, the REST API should check the DB to calculate the metrics.

**Comment by deruelle** [comment]
@forever3000 were you able to make more progress on this ?  I think we should provide this as part of the Usage API actually as an additional field

**Comment by forever3000** [comment]
Hi @deruelle ,
Sorry for stuck a time. I will start again.

**Comment by forever3000** [comment]
Hi @gvagenas ,

I already created maxConcurrentCallsLastHour and maxConcurrentCallsLastDay based on start time and end time of CDR querying from DB. About the one for a specific day, we should have one for a period (start date - end date) instead. 
Currently, I modified restcomm_with_data script to run testsuite however time is going so we can not fix start time, end time in script. Could you give me some advices. 

Thanks
